aiohttp/web_request.py

```diff
@@ -510,8 +511,26 @@ def text(self):
         return bytes_body.decode(encoding)
 
     @asyncio.coroutine
-    def json(self, *, loads=json.loads):
-        """Return BODY as JSON."""
+    def json(self, *, loads=json.loads, allowed_type=None):
+        """ Return BODY as JSON. If the `Content-Type` does not equal
+        allowed_type, then the request is rejected.
+        """
+
+        if allowed_type is None:  # do not check the content_type
+            pass
+        elif isinstance(allowed_type, str):
+            # parse the content_type
+            content_type = self.content_type
+            (mime_type, subtype, suffix, _) = parse_mimetype(content_type)
+            content_type = mime_type + '/' + subtype
+            if suffix != '':
+                content_type += '+' + suffix
+
+            if allowed_type != content_type:
+                raise HTTPNotAcceptable()
```

Comment:
So, what do you guys think now? Should the message be passed as an argument? I think the error should be raised directly since that's exactly what I want for a JSON API in most cases and I think we've reached the agreement that for special requirements one should write a middleware or catch the exception as we discussed before.